theano/gpuarray/multinomial.py

```diff
@@ -327,6 +327,7 @@ def gpu_kernels(self, node, name):
 
                     if (! %(replace)s )
                     {
+                        __syncthreads();
```

Comment:
In the end, I think scaling the uniform sample rather than the distribution would remove the need for LID_1 > 1, so we can avoid the conflicts. I'll do that soon.